src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp

```diff
@@ -80,6 +80,22 @@ bool FlightTaskManualAltitude::activate()
 	return ret;
 }
 
+void FlightTaskManualAltitude::_updateConstraints()
+{
+	// Altitude above ground is by default just the negation of the current local position in D-direction.
+	float alt_above_ground = -_position(2);
+
+	if (PX4_ISFINITE(_dist_to_bottom)) {
+		// We have a valid distance to ground measurement
+		alt_above_ground = _dist_to_bottom;
+	}
+
+	// Below ALT1, limit down speed
+	_constraints.speed_down = (alt_above_ground < MPC_LAND_ALT1.get()) ? MPC_LAND_SPEED.get() : _min_speed_down;
```

Comment:
I would gradually adjust the speed between the two altitudes. That's how it was before and it results in very good user experience even with a quite big difference between the limits. See https://github.com/PX4/Firmware/pull/11540/files#diff-4d17e8ffdb52e603266999d7af0f3a82R282